File: lib/py/uapi/internal/validation/ValidateInteger.py
from typing import object, list, Union
from uapi.internal.types.UInteger import _INTEGER_NAME
from uapi.internal.validation.ValidationFailure import ValidationFailure


def validate_integer(value: object) -> list['ValidationFailure']:
    from uapi.internal.validation.GetTypeUnexpectedValidationFailure import get_type_unexpected_validation_failure

    if isinstance(value, (int)) and not isinstance(value, (bool, float)):
        if (value > 2**63-1 or value < -(2**63)):
            return [ValidationFailure([], "NumberOutOfRange", {})]
        else:
            return []

    return get_type_unexpected_validation_failure([], value, _INTEGER_NAME)

File: lib/ts/uapi/internal/validation/ValidateInteger.ts
import { ValidationFailure } from 'uapi/internal/validation/ValidationFailure';
import { getTypeUnexpectedValidationFailure } from 'uapi/internal/validation/GetTypeUnexpectedValidationFailure';
import { integerName } from 'uapi/internal/types/UInteger';

export function validateInteger(value: any): ValidationFailure[] {
    if (typeof value === 'number' && Number.isInteger(value)) {
        if (value === 9223372036854776000 || value === -9223372036854776000) {
            return [
                new ValidationFailure([], 'NumberOutOfRange', {}),
                new ValidationFailure([], 'NumberTruncated', {}),
            ];
        }
        return [];
    } else {
        return getTypeUnexpectedValidationFailure([], value, integerName);
    }
}
